pkg/ndm-grpc/server/services/listBlockDevice.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2019 The OpenEBS Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"github.com/openebs/node-disk-manager/pkg/apis/openebs/v1alpha1"
+	protos "github.com/openebs/node-disk-manager/pkg/ndm-grpc/protos/ndm"
+
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openebs/node-disk-manager/cmd/ndm_daemonset/controller"
+
+	"github.com/openebs/node-disk-manager/pkg/hierarchy"
+	"github.com/openebs/node-disk-manager/pkg/ndm-grpc/server"
+	"github.com/sirupsen/logrus"
+)
+
+// Node helps in using types defined in Server
+type Node struct {
+	server.Node
+}
+
+// NewNode is a constructor
+func NewNode(l *logrus.Logger) *Node {
+	return &Node{server.Node{Log: l}}
+}
+
+type disks []protos.BlockDevice
+
+// ListBlockDevices gives the status of iSCSI service
+func (n *Node) ListBlockDevices(ctx context.Context, null *protos.Null) (*protos.BlockDevices, error) {
+
+	n.Log.Info("Listing block devices")
+
+	ctrl, err := controller.NewController()
+	if err != nil {
+		n.Log.Errorf("Error creating a controller %v", err)
+		return nil, status.Errorf(codes.NotFound, "Namespace not found")
+	}
+
+	err = ctrl.SetControllerOptions(controller.NDMOptions{ConfigFilePath: "/host/node-disk-manager.config"})
+	if err != nil {
+		n.Log.Errorf("Error setting config to controller %v", err)
+		return nil, status.Errorf(codes.Internal, "Error setting config to controller")
+	}
+
+	blockDeviceList, err := ctrl.ListBlockDeviceResource(false)
+	if err != nil {
+		n.Log.Errorf("Error listing block devices %v", err)
+		return nil, status.Errorf(codes.Internal, "Error fetching list of disks")
+	}
+
+	if len(blockDeviceList.Items) == 0 {
+		n.Log.Info("No items found")
+	}
+
+	blockDevices := make([]*protos.BlockDevice, 0)
+
+	parentNames, err := GetParentDisks(n, blockDeviceList)
```

Comment:
Can a comment be provided here, what we are trying to. I am not able to understand the output format.